### DISCUSSION TITLE
why dose not support async operations in dubbo provider side

### DISCUSSION THREAD

**Discussion by alalag1** [open-telemetry/opentelemetry-java-instrumentation#10362] [Q&A]
Why is it that in the Dubbo instrumentation, there is a support for async operations only on the consumer side, and not on the provider side? Is there any special consideration behind this?

https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/f52a3b404a68f93a257c5d97f6692a2763646303/instrumentation/apache-dubbo-2.7/library-autoconfigure/src/main/java/io/opentelemetry/instrumentation/apachedubbo/v2_7/TracingFilter.java#L57-L71

**Comment by laurit** [comment]
At this point I don't think there is anybody who can tell why it was done so. If you so wish you can try tracking down the original author of the code, but keep in mind that it was written a couple of years ago. If this is something that you need then feel free to submit a PR that fixes this along with tests.

**Comment by trask** [comment]
cc @tydhot @zmapleshine @yingziisme
